src/ApplicationInsights.ServiceFabric.Native/Net45/CorrelatingServiceRemotingClient.cs

```diff
@@ -0,0 +1,155 @@
+
+        /// <summary>
+        /// Gets or Sets the name of the listener in the replica or instance to which the client is connected to.
+        /// </summary>
+        public string ListenerName { get => this.InnerClient.ListenerName; set => this.InnerClient.ListenerName = value; }
+
+        /// <summary>
+        /// Gets or Sets the service endpoint to which the client is connected to.
+        /// </summary>
+        public ResolvedServiceEndpoint Endpoint { get => this.InnerClient.Endpoint; set => this.InnerClient.Endpoint = value; }
+
+        /// <summary>
+        /// Gets the inner client which this client wraps.
+        /// </summary>
+        public IServiceRemotingClient InnerClient { get; private set; }
+
+        /// <summary>
+        /// Sends a message to the service and gets a response back. The correlation id and context are sent along with
+        /// the message as message headers.
+        /// </summary>
+        /// <param name="messageHeaders">Message headers.</param>
+        /// <param name="requestBody">Message body.</param>
+        /// <returns>Response body.</returns>
+        public Task<byte[]> RequestResponseAsync(ServiceRemotingMessageHeaders messageHeaders, byte[] requestBody)
+        {
+            return SendAndTrackRequestAsync(messageHeaders, requestBody, () => this.InnerClient.RequestResponseAsync(messageHeaders, requestBody));
+        }
+
+        /// <summary>
+        /// Sends a one-way message to the service. The correlation id and context are sent along with the message
+        /// as message headers.
+        /// </summary>
+        /// <param name="messageHeaders">Message headers.</param>
+        /// <param name="requestBody">Message body.</param>
+        public void SendOneWay(ServiceRemotingMessageHeaders messageHeaders, byte[] requestBody)
+        {
+            SendAndTrackRequestAsync(messageHeaders, requestBody, () =>
+            {
+                this.InnerClient.SendOneWay(messageHeaders, requestBody);
+                return Task.FromResult<byte[]>(null);
+            }).Forget();
+        }
+
+        private async Task<byte[]> SendAndTrackRequestAsync(ServiceRemotingMessageHeaders messageHeaders, byte[] requestBody, Func<Task<byte[]>> doSendRequest)
+        {
+            string methodName = this.methodNameProvider.GetMethodName(messageHeaders.InterfaceId, messageHeaders.MethodId);
+
+            // Weird case, just use the numerical id as the method name
+            if (string.IsNullOrEmpty(methodName))
+            {
+                methodName = messageHeaders.MethodId.ToString();
+            }
+
+            // Since service remoting doesn't really have an URL like HTTP URL, we will fake one up here containing the service URI with the interface id and
+            // method id of the remote method
+            string operationName = this.serviceUri.AbsoluteUri + "/" + methodName;
```

Comment:
the operation name should not contain full URI, just the path.
Data should have absolute url
Target should be the host name, not the url,

E.g. check the AI code: https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/develop/Src/DependencyCollector/Shared/Implementation/HttpCoreDiagnosticSourceListener.cs


```
           Uri requestUri = request.RequestUri;
            var resourceName = request.Method.Method + " " + requestUri.AbsolutePath;

            DependencyTelemetry telemetry = new DependencyTelemetry();

            // properly fill dependency telemetry operation context: OperationCorrelationTelemetryInitializer initializes child telemetry
            telemetry.Context.Operation.Id = currentActivity.RootId;
            telemetry.Context.Operation.ParentId = currentActivity.ParentId;
            telemetry.Id = currentActivity.Id;
            foreach (var item in currentActivity.Baggage)
            {
                if (!telemetry.Context.Properties.ContainsKey(item.Key))
                {
                    telemetry.Context.Properties[item.Key] = item.Value;
                }
            }
            
            this.client.Initialize(telemetry);

            telemetry.Timestamp = currentActivity.StartTimeUtc;
            telemetry.Name = resourceName;
            telemetry.Target = requestUri.Host;
            telemetry.Type = RemoteDependencyConstants.HTTP;
            telemetry.Data = requestUri.OriginalString;
            telemetry.Duration = currentActivity.Duration;
```